db/compaction_job.cc

```diff
@@ -747,6 +747,14 @@ void CompactionJob::ProcessKeyValueCompaction(SubcompactionState* sub_compact) {
       shutting_down_));
   auto c_iter = sub_compact->c_iter.get();
   c_iter->SeekToFirst();
+  if (c_iter->Valid()) {
+    // ShouldStopBefore() maintains state based on keys processed so far. The
+    // compaction loop always calls it on the "next" key, thus won't tell it the
+    // first key. So we do that here.
+    bool should_stop = sub_compact->ShouldStopBefore(
+      c_iter->key(), sub_compact->current_output_file_size);
+    assert(!should_stop);
```

Comment:
I think so, because ShouldStopBefore() checks the amount of overlap with grandparent level, excluding the current key. In this case, the current key is the first key, so there'll be no overlap.

Even if ShouldStopBefore() changes in the future to do more checks, I still think it'd be a bug for it to try ending a file before (i.e., excluding) the first key.